gusto/eady_diagnostics.py

```diff
@@ -54,15 +35,9 @@ def compute(self, state):
         return self.field(state.mesh).interpolate(potential)
 
 
-class CompressibleEadyPotentialEnergy(DiagnosticField):
+class CompressibleEadyPotentialEnergy(Energy):
     name = "CompressibleEadyPotentialEnergy"
 
-    def field(self, mesh):
-        if hasattr(self, "_field"):
-            return self._field
-        self._field = Function(FunctionSpace(mesh, "DG", 0), name=self.name)
-        return self._field
-
     def compute(self, state):
         x, y, z = SpatialCoordinate(state.mesh)
         g = state.parameters.g
```

Comment:
This is unique to the Compressible Eady problem which needs the extra term - cp*Pi0*theta.